app/org/sagebionetworks/bridge/models/schedules/Activity.java

```diff
@@ -1,49 +1,132 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public class Activity {
-
-    private final String label;
-    private final ActivityType activityType;
-    private final String ref;
+/**
+ * An activity we wish a study participant to do. The two main types of activities are 
+ * tasks in the application (such as taking a tapping test), and surveys to be filled 
+ * out and returned to the sever.
+ * 
+ * In schedules, activities are set with surveys. In tasks, these are "resolved" to point 
+ * to a specific survey, and a survey response object is also added to the activity. This includes 
+ * an endpoint where answers to the survey can be submitted.
+ */
+public final class Activity implements BridgeEntity {
     
+    private static final ActivityValidator VALIDATOR = new ActivityValidator();
+    
+    private String label;
+    private String labelDetail;
+    private TaskReference task;
+    private SurveyReference survey;
+    private SurveyResponseReference response;
+    private ActivityType activityType;
+
     @JsonCreator
-    public Activity(@JsonProperty("label") String label, @JsonProperty("ref") String ref) {
+    private Activity(@JsonProperty("label") String label, @JsonProperty("labelDetail") String labelDetail, 
+        @JsonProperty("task") TaskReference task, @JsonProperty("survey") SurveyReference survey, 
+        @JsonProperty("surveyResponse") SurveyResponseReference response) {
         this.label = label;
-        this.ref = ref;
-        if (ref == null) {
-            this.activityType = null;
-        } else {
-            this.activityType = SurveyReference.isSurveyRef(ref) ? ActivityType.SURVEY : ActivityType.TASK;    
-        }
+        this.labelDetail = labelDetail;
+        this.survey = survey;
+        this.task = task;
+        this.survey = survey;
+        this.response = response;
+        this.activityType = (task != null) ? TASK : SURVEY;
     }
     
-    public String getLabel() {
+    public String getLabel() { 
         return label;
     }
-
+    public String getLabelDetail() {
+        return labelDetail;
+    }
     public ActivityType getActivityType() {
         return activityType;
     }
-
-    public String getRef() {
-        return ref;
+    public TaskReference getTask() {
+        return task;
     }
-
     public SurveyReference getSurvey() {
-        return SurveyReference.isSurveyRef(ref) ? new SurveyReference(ref) : null;
+        return survey;
+    }
+    public SurveyResponseReference getSurveyResponse() {
+        return response;
+    }
+    /**
+     * This property is maintained for backwards compatibility, but clients should now look for 
+     * metadata information, including links, in the survey, surveyResponse or task property 
+     * objects.
+     * @return
+     */
+    public String getRef() { 
+        if (task != null) {
+            return task.getIdentifier();
+        } else if (survey != null) {
```

Comment:
I think this if-else is another indicator of the TaskActivity and SurveryActivity subtypes.